Using PQL to grab DNS ipaddress

Nekototori opened the pull request:
Since joining a domain requires the dns resolver to find the MS DNS records, this requires looking up the ipaddress of the AD DNS server, and then setting it as the DNS server.

This assumes that the AD server for the class always has the certname of adserver.  An alternative would be to do a lookup of a class that only ever gets assigned to the ad server.

Once DSC is fully integrated, it'd be best to use an exported resource from that, but this solves the issue until then.


This also updates the unless to check for the ad ipaddress, since as long as that is there we don't care so much about the rest.

Nekototori commented:
Added validation, a default value, and fixed the syntax, both the missing puppetdb_query and the '.

samuelson commented:
@binford2k I'm fine with this as long as it's well tested before our next Windows course.

@Nekototori if you're so inclined, I think a better solution would be to roll out the DSC code. It was initially commented out because it depended on an unreleased module, which has now been released. I'm not sure when the EDU team will have time to prioritize that work, probably not this quarter.

Nekototori commented:
I rebased this PR with the master.  Not sure if you want me to do anything with the old commits.  The only one that matters at this point is f12c892b221267cab301dde1fcfd20424e6a1027

Once I was in a classroom environment I traced the whole code and saw it was already dealing with an exported resource, so the issue ended up being how the exec was called, specifically in the classroom environment.  I've updated the exec to apply the ip address of the ad server to all interfaces starting with the name of Ethernet0.
